webapp/app/data_access/form_data_controller.py

```diff
@@ -0,0 +1,49 @@
+import redis
+import redislite
+from app.config import Config
+from pydantic import MissingError
+
+
+class FormDataController:
+
+    _instance = None
+    _redis_connection = None
+
+    def __new__(cls):
+        """Singleton creation so that the redis connection is not recreated."""
+        if cls._instance is None:
+            cls._instance = super(FormDataController, cls).__new__(cls)
+            cls._redis_connection = cls.get_redis_connection()
+        return cls._instance
```

Comment:
Could you maybe add a test for that? Furthermore, shouldn't get_redis_connection than be an anonymous function? Because we do not want to get a different connection?

Reply:
Have a look Julian, I replaced with a simple ternary operator, rather than using an anonymous function, wdyt?